File: frontend/src/types/order.ts
export type OrderStatus = 'pending' | 'confirmed' | 'out_for_delivery' | 'delivered' | 'cancelled';

export interface OrderItem {
  id: string;
  order_id: string;
  product_id: string;
  quantity: string;
  unit_price: string;
  subtotal: string;
  created_at: string;
}

export interface Order {
  id: string;
  user_id: string;
  delivery_area_id: string;
  delivery_slot_id: string;
  delivery_date: string;
  delivery_address: string;
  status: OrderStatus;
  total_amount: string;
  created_at: string;
  updated_at: string;
}

export interface OrderWithItems extends Order {
  items: OrderItem[];
}

File: backend/internal/model/order.go
package model

import "time"

const (
	OrderStatusPending        = "pending"
	OrderStatusConfirmed      = "confirmed"
	OrderStatusOutForDelivery = "out_for_delivery"
	OrderStatusDelivered      = "delivered"
	OrderStatusCancelled      = "cancelled"
)

type Order struct {
	ID              string    `db:"id" json:"id"`
	UserID          string    `db:"user_id" json:"user_id"`
	DeliveryAreaID  string    `db:"delivery_area_id" json:"delivery_area_id"`
	DeliverySlotID  string    `db:"delivery_slot_id" json:"delivery_slot_id"`
	DeliveryDate    time.Time `db:"delivery_date" json:"delivery_date"`
	DeliveryAddress string    `db:"delivery_address" json:"delivery_address"`
	Status          string    `db:"status" json:"status"`
	TotalAmount     string    `db:"total_amount" json:"total_amount"` // NUMERIC as string to avoid float rounding on currency
	CreatedAt       time.Time `db:"created_at" json:"created_at"`
	UpdatedAt       time.Time `db:"updated_at" json:"updated_at"`
}
